File: assignment1_q2.c
#include <stdio.h>

int main() {
	int i;
	int n;
	int factorial = 1;

	printf("Enter a number n to find the factorial of n: "); 
	scanf("%d", &n);

	if (n<0){
		printf("You have entered a negative number.");
	}

	else if (n == 0){
		printf("0! = 1");
	}

	else {
		printf("%d! = ", n);
		for (i = 1; i <= n; ++i){
			factorial = factorial * i;
			if(i<n){
				printf("%d X ", i);
			}
			else{
				printf("%d", i);
			}
		}
		printf(" = %d", factorial);
	}

	return 0;
}

File: assignment1_q2.py
x = input("Enter a number n to find the factorial of n: ")
factorial = 1

if(int(x)<0):
    print("You have entered a negative number.")
elif(int(x) == 0):
    print("0! = 1")
else:
    print(str(x) + "! = ", end = '')
    for i in range(1, int(x)+1):
        factorial = factorial * i
        if(i<int(x)):
            print(str(i) + " X ", end = '')
        else:
            print(i, end = '')
    print(" = " + str(factorial))

